### PR TITLE
Removed 'using namespace llvm' in header files

### PR DESCRIPTION
**Pull request by trenouf**
It has been agreed that "using namespace" in a header file is a bad
idea, even inside a namespace. This commit removes them.

Change-Id: Ia1ecd62d97a822e382cc1f07192d19f135d73230

### CODE REVIEW

**Comment by trenouf** [comment]
This messes up some function parameter alignment. But that is about to be reformatted by the coding style change anyway.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 8580c4fe5b7369b60996dbe82edca95bf1fc241c
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev bb7398b1c7509a30ba4f7d947b5ad5549a4d58db</li>
<li>SPVGEN: dev dc2b146de03ef54f188172d8fa2f9be708cbfaca</li>
<li>XGL: dev ffd39443fec0ce821b7702173eba7184855c7b5e</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 1ebc431ac299fa294a4e730432520d37f326843c</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 955/955 (100.0%)</li>
<li>Failed: 0/955 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103137/103137 (100.0%)</li>
<li>Failed: 0/103137 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 103193/103193 (100.0%)</li>
<li>Failed: 0/103193 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by trenouf** [comment]
> I think we should write this to the coding standard doc, explicitly forbidding "using namespace XXX" in header files.

Good idea.

File: lgc/patch/llpcIntrinsDefs.h [hunk 1]
```diff
@@ -67,9 +67,9 @@ enum AddrSpace
 {
     ADDR_SPACE_GLOBAL             = 1,          // Global memory
     ADDR_SPACE_LOCAL              = 3,          // Local memory
-    ADDR_SPACE_CONST              = 4,          // Constant memory
+    ADDR_SPACE_CONST              = 4,          // llvm::Constant memory
     ADDR_SPACE_PRIVATE            = 5,          // Private memory
-    ADDR_SPACE_CONST_32BIT        = 6,          // Constant 32-bit memory
+    ADDR_SPACE_CONST_32BIT        = 6,          // llvm::Constant 32-bit memory
```
Comment (kuhar) [inline lgc/patch/llpcIntrinsDefs.h:72]: unintended?

**Comment by trenouf** [comment]
V2: Removed some accidental llvm:: additions in comments.

File: lgc/patch/llpcIntrinsDefs.h (hunk 1 above)
Comment (trenouf) [reply lgc/patch/llpcIntrinsDefs.h]: Oops thanks. Now fixed.

**Comment by trenouf** [comment]
> Thanks, Tim.
> 
> Did you use clang-tidy to do the refactoring? I'm curious to learn how easy it is to do these mass changes.

No. Removing the "using namespaces" was a script, and fixing up the resulting errors was mostly manual with some global search-and-replace in my editor.

The forthcoming coding standards change is mostly done with clang-tidy though.

**Comment by trenouf** [comment]
As I only changed three comments against the first version of this change, I'm going to land it now without waiting for its test results.

**Comment by trenouf** [comment]
> Do you think we should also write a rule in the coding standard doc that allow developers to "using namespace XXX" in source files. For "std", it seems we always avoid this. For other namespaces, we always do this, such as "using namespace Pal/llvm/spv".

See https://llvm.org/docs/CodingStandards.html#do-not-use-using-namespace-std in the LLVM coding standards doc. Maybe we could explicitly point it out in our doc.

**Comment by amdrexu** [comment]
> See https://llvm.org/docs/CodingStandards.html#do-not-use-using-namespace-std in the LLVM coding standards doc. Maybe we could explicitly point it out in our doc.

Thank you, I see. I agree we can point this out in the section " `using namespace`".

**Comment by amdvlk-admin** [comment]
### Test summary for commit 109b24770179fb4c102b82fcdde80019e8ff3aa2
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev bb7398b1c7509a30ba4f7d947b5ad5549a4d58db</li>
<li>SPVGEN: dev dc2b146de03ef54f188172d8fa2f9be708cbfaca</li>
<li>XGL: dev ffd39443fec0ce821b7702173eba7184855c7b5e</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 1ebc431ac299fa294a4e730432520d37f326843c</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 955/955 (100.0%)</li>
<li>Failed: 0/955 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103137/103137 (100.0%)</li>
<li>Failed: 0/103137 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 103193/103193 (100.0%)</li>
<li>Failed: 0/103193 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**